hibernate-core/src/main/java/org/hibernate/hql/internal/ast/util/JoinProcessor.java

```diff
@@ -171,9 +172,20 @@ private void collectReferencedTables(ASTIterator iterator, Set<String> result) {
 				FromReferenceNode fromReferenceNode = (FromReferenceNode) node;
 				String[] tables = fromReferenceNode.getReferencedTables();
 				if ( tables != null ) {
-					for ( String table : tables ) {
```

Comment:
Again, on purpose to avoid the array list

Reply:
Switched to `Collections.addAll` which also avoids the creation of the intermediate array list. (Aligned with the transition introduced in #2995 )